### PR TITLE
[v0.1] code polishing: tools/filter.py

### PR DESCRIPTION
**Pull request by trz42**
Code polishing pass on `tools/filter.py` for release v0.1:

- imports: same categorisation of imports
- docstrings: slightly rephrased some + added a missing information
- comments: improved some comments

### CODE REVIEW

File: tools/filter.py [hunk 1]
```diff
@@ -55,30 +73,47 @@ def __init__(self, filter_string):
...
-            # 'r' and 're')
+            # prefix (e.g., 'repository' and 'request' would have the same
+            # prefixes 'r' and 're')
             if cis.startswith(component):
                 full_component = cis
                 break
         if full_component:
             log(f"processing component {component}")
-            # if full_component == architecture replace - with / in pattern
+            # if full_component == 'architecture' replace '-' with '/' in
```
Comment (boegel) [inline tools/filter.py:115]: `replace '-' with '/' in pattern when using 'architecture' filter component`

File: tools/filter.py [hunk 2]
```diff
@@ -124,6 +169,12 @@ def remove_filter(self, component, pattern):
     def to_string(self):
         """
         Convert filters to string
+
+        Args:
+            No arguments
+
+        Returns:
+            string containing whitespace separated filters
```
Comment (boegel) [inline tools/filter.py:177]: `string containing filters separated by whitespace`

**Review by boegel — CHANGES_REQUESTED** [review]

File: tools/filter.py [hunk 3]
```diff
@@ -88,10 +123,17 @@ def add_filter(self, component, pattern):
...
         Args:
-            filter_string (string): filter provided as command:pattern
+            filter_string (string): filter provided as command:pattern string
+
+        Returns:
+            None (implicitly)
+
+        Raises:
+           EESSIBotActionFilterError: raised if filter_string does not conform
+               to 'component:pattern' format or pattern is empty
```
[the remark this answers is not in the document]
Comment (trz42) [reply tools/filter.py:136]: Args description was wrong. Nice catch!

File: tools/filter.py [hunk 4]
```diff
@@ -55,30 +73,47 @@ def __init__(self, filter_string):
...
+        # check if component is supported (i.e., it is a prefix of one of the
+        # elements in FILTER_COMPONENTS)
         full_component = None
         for cis in FILTER_COMPONENTS:
             # NOTE the below code assumes that no two filter share the same
-            # prefix (e.g., repository and request would have the same prefixes
-            # 'r' and 're')
+            # prefix (e.g., 'repository' and 'request' would have the same
+            # prefixes 'r' and 're')
             if cis.startswith(component):
```
[the remark this answers is not in the document]
Comment (trz42) [reply tools/filter.py:110]: I will open an issue. Ideally the procedure would check if the prefix is unique, refuses to process it if not, and provides some guidance. For example, _Component 're' is ambiguous (could be 'repository' or 'request'). Resent the command with a longer unambiguous prefix._

[the remark this answers is not in the document]
Comment (boegel) [reply tools/filter.py:109]: issue opened at #210

**Review by boegel — APPROVED** [review]

**Pull request merged**